src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -608,7 +620,26 @@ case class IndexLogEntryTag[T](name: String)
 object IndexLogEntry {
   val VERSION: String = "0.1"
 
+  val HYPERSPACE_PROJECT_VERSION: String = "hyperspaceVersion"
+
   def schemaString(schema: StructType): String = schema.json
+
+  /*
+   * Default IndexLogEntry Properties
+   */
+  def apply(name: String,
+            derivedDataset: CoveringIndex,
+            content: Content,
+            source: Source
```

Comment:
seems like this doesn't help either since I can create IndexLogEntry directly? In that case, `create` is just fine supporting all fields?

Reply:
Won't you be able to create `IndexLogEntry` directly with create method as well ? 

This apply signature signifies that it is a default constructor for `IndexLogEntry` when you specify no properties and include default properties. Whereas if you specify properties then specify all properties, the create method seems more hacky to me ?

Reply:
But it doesn't add any value of enforcing the requirement.

what we want is not supported by the language (unless we restructure this class with `sealed abstract` - there are few examples on the web), so anything we are doing is not the ideal approach (if you mean hacky by that). Also, I am not inventing create method approach: https://github.com/apache/spark/blob/56e664c7179eadeb5134b4418f3aaa6a9d742ef6/sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala#L169

Reply:
Another downside of this approach is that if you now want to add a prop on the caller side, you have to always manually add the version prop as well.

Reply:
Okay, switched to create format that is commonly used in the project.